build-tools/build.sh

```diff
@@ -0,0 +1,223 @@
+echo "Library name found: $library_name"
+
+echo "Version before setting postfix: $version"
+echo "SHA before setting postfix: $sha"
+
+if [ "$INPUT_NO_COMMIT_HASH" == "true" ]; then
+    postfix="${version}"
+else
+    postfix="${version}+${sha}"
+fi
+echo "Postfix after setting: $postfix"
+
+name="$library_name@$postfix"
+echo "name=$name" >>"$GITHUB_OUTPUT"
+echo "Name found: $name"
+
+echo "::endgroup::"
+# ----------------
+# BUILDING PROJECT
+# ----------------
+# Pause errors
+set +e
+make clean $make_args
+ERR_OUTPUT=$(mktemp)
+STD_OUTPUT=$(mktemp)
+# Set IS_LIBRARY to 0 to build the project if $template is 1
+if (($template == 1)); then
+    echo "::group::Building ${name}"
+    echo "Setting IS_LIBRARY to 0"
+    sed -i "s/^IS_LIBRARY:=.*\$/IS_LIBRARY:=0/" Makefile
+fi
+
+# Actual build
+start_build_time=$SECONDS
+make quick $make_args 2>$ERR_OUTPUT | tee $STD_OUTPUT
+build_time=$((SECONDS - $start_build_time))
+
+# Set IS_LIBRARY back to 1 if $template was 1
+if (($template == 1)); then
+    echo "Setting IS_LIBRARY back to 1"
+    sed -i "s/^IS_LIBRARY:=.*\$/IS_LIBRARY:=1/" Makefile
+    echo "::endgroup::"
+fi
+
+STD_EDITED_OUTPUT=$(mktemp)
+# Remove ANSI color codes from the output
+sed -e 's/\x1b\[[0-9;]*m//g' $STD_OUTPUT >$STD_EDITED_OUTPUT
+
+if [ -s "$ERR_OUTPUT" ]; then
+    error_output=$(cat "$ERR_OUTPUT")
+    norm_output=$(cat "$STD_EDITED_OUTPUT")
+    echo "# 🛑 Build Failed" >>$GITHUB_STEP_SUMMARY
+    echo "#### 📄 Error Output" >>$GITHUB_STEP_SUMMARY
+    echo "Build failed in $build_time seconds" >>$GITHUB_STEP_SUMMARY
+    echo "Total Build Script Runtime: $(($SECONDS - $script_start_time)) seconds" >>$GITHUB_STEP_SUMMARY
+    echo "<details><summary>Click to expand</summary>" >>$GITHUB_STEP_SUMMARY
+    echo "" >>$GITHUB_STEP_SUMMARY
+    echo "" >>$GITHUB_STEP_SUMMARY
+    echo "\`\`\`" >>$GITHUB_STEP_SUMMARY
+    echo "$norm_output" >>$GITHUB_STEP_SUMMARY
+    echo "\`\`\`" >>$GITHUB_STEP_SUMMARY
+    echo "</details>" >>$GITHUB_STEP_SUMMARY
+    exit 1
+fi
+
+# -----------------
+# CREATING TEMPLATE
+# -----------------
+
+set -e # Enable exiting on error
+
+if (($template == 1)); then
+    echo "::group::Updating Makefile"
+
+    sed -i "s/^VERSION:=.*\$/VERSION:=${postfix}/" Makefile
+
+    cat Makefile
+
+    echo "::endgroup::"
+
+    echo "::group::Creating ${name} template"
+
+    make template $make_args
+
+    echo "::endgroup::"
+
+    # --------------
+    # UNZIP TEMPLATE
+    # --------------
+
+    echo "::group::Unzipping template"
+
+    unzip -o $name -d template # Unzip the template
+
+    echo "::endgroup::"
+
+fi
+
+# ---------------------------
+# ADDING VERSION, LICENSE
+# AND README TO THE TEMPLATE
+# FOLDER
+# ---------------------------
+if [[ "$INPUT_COPY_README_AND_LICENSE_TO_INCLUDE" == "true" ]]; then
+    if [[ "$INPUT_LIB_FOLDER_NAME" != "" ]]; then
+        echo "::group::Adding version, license and readme to the template folder"
```

Comment:
shouldn't the `::group::` and `::endgroup::` go outside this if statement?

Reply:
No, because if that is false then there's no point in showing the group in the output.

Reply:
in the first if statement sure, but after that it is always printed